node/options.go

```diff
@@ -542,6 +542,18 @@ func WithMinTransactionVersion(minTransactionVersion uint32) Option {
 	}
 }
 
+// WithMaxTransactionValueLength sets up the maximum length for the transaction's value field
+func WithMaxTransactionValueLength(maxLength int) Option {
+	return func(n *Node) error {
+		if maxLength == 0 {
+			return ErrInvalidTransactionVersion
```

Comment:
Change the error according to this method. Anyway I'm not sure if this check is necessary or if it is, we should check with <= as the maxLength is of int type, so it could have a negative value

Reply:
this check is useful in case of default values (0). extended the protection also for negative values